java/core/src/java/org/apache/orc/geospatial/BoundingBox.java

```diff
@@ -0,0 +1,374 @@
+   * The X dimension is considered valid if neither bound contains NaN.
+   *
+   * @return true if the X dimension is valid, false otherwise.
+   */
+  public boolean isXValid() {
+    return !(Double.isNaN(xMin) || Double.isNaN(xMax));
+  }
+
+  /**
+   * Checks if the Y dimension of the bounding box is valid.
+   * The Y dimension is considered valid if neither bound contains NaN.
+   *
+   * @return true if the Y dimension is valid, false otherwise.
+   */
+  public boolean isYValid() {
+    return !(Double.isNaN(yMin) || Double.isNaN(yMax));
+  }
+
+  /**
+   * Checks if the Z dimension of the bounding box is valid.
+   * The Z dimension is considered valid if none of the bounds contain NaN.
+   *
+   * @return true if the Z dimension is valid, false otherwise.
+   */
+  public boolean isZValid() {
+    return !(Double.isNaN(zMin) || Double.isNaN(zMax));
+  }
+
+  /**
+   * Checks if the M dimension of the bounding box is valid.
+   * The M dimension is considered valid if none of the bounds contain NaN.
+   *
+   * @return true if the M dimension is valid, false otherwise.
+   */
+  public boolean isMValid() {
+    return !(Double.isNaN(mMin) || Double.isNaN(mMax));
+  }
+
+  /**
+   * Checks if the bounding box is empty in the X / Y dimension.
+   *
+   * @return true if the bounding box is empty, false otherwise.
+   */
+  public boolean isXYEmpty() {
+    return isXEmpty() || isYEmpty();
+  }
+
+  /**
+   * Checks if the bounding box is empty in the X dimension.
+   *
+   * @return true if the X dimension is empty, false otherwise.
+   */
+  public boolean isXEmpty() {
+    return Double.isInfinite(xMin) && Double.isInfinite(xMax);
+  }
+
+  /**
+   * Checks if the bounding box is empty in the Y dimension.
+   *
+   * @return true if the Y dimension is empty, false otherwise.
+   */
+  public boolean isYEmpty() {
+    return Double.isInfinite(yMin) && Double.isInfinite(yMax);
+  }
+
+  /**
+   * Checks if the bounding box is empty in the Z dimension.
+   *
+   * @return true if the Z dimension is empty, false otherwise.
+   */
+  public boolean isZEmpty() {
+    return Double.isInfinite(zMin) && Double.isInfinite(zMax);
+  }
+
+  /**
+   * Checks if the bounding box is empty in the M dimension.
+   *
+   * @return true if the M dimension is empty, false otherwise.
+   */
+  public boolean isMEmpty() {
+    return Double.isInfinite(mMin) && Double.isInfinite(mMax);
+  }
+
+  /**
+   * Expands this bounding box to include the bounds of another box.
+   * After merging, this bounding box will contain both its original extent
+   * and the extent of the other bounding box.
+   *
+   * @param other the other BoundingBox whose bounds will be merged into this one
+   */
+  public void merge(BoundingBox other) {
+    if (!valid) {
+      return;
+    }
+
+    // If other is null or invalid, mark this as invalid
+    if (other == null || !other.valid) {
+      valid = false;
+      resetBBox();
```

Comment:
If we want to have `valid=false`, we need to change the order of the above statements because `resetBBox` will reset `valid` too. Is there any documentation about this behavior (which is merging to invalid BoundingBox), @ffacs ?

**FROM**
```
      valid = false;
      resetBBox();
```

**TO**
```
      resetBBox();
      valid = false;
```

Reply:
Thank you for the decision and update, @ffacs .